### ISSUE TITLE
Subdirectory/relative path extraction

### ISSUE THREAD

**Issue by tomasfejfar** [eloquent/pathogen#39]
Is it currently possible to do a "reverse resolving"? Get relative subdirectory path from base path and absolute path? 

```php
$basePath = '/var/www/';
$absolutePath = FileSystemPath::fromString('/var/www/project/public/index.php');
$relativePath = $absolutePath->getRelativePath($basePath);
// project/public/index.php
```

**Comment by ezzatron** [comment]
Yes, this is possible with [relativeTo()](https://github.com/eloquent/pathogen/blob/66dec97/src/AbsolutePathInterface.php#L56).

However, if you're translating HTTP request paths into paths on disk, please be aware of the various [path traversal attacks](https://www.owasp.org/index.php/Path_Traversal) that you might open yourself up to. You might want to look at Guzzle's [UriResolver](https://github.com/guzzle/psr7#reference-resolution), to do some sanitation/validation before your request path even makes it into a file system path.
